caps/pip.md

```diff
@@ -0,0 +1,431 @@
+["txRelay”,  tx_relay]
+["flowControl/BL", flow_control_bl]
+["flowControl/MRC", flow_control_mrc]
+["flowControl/MRR", flow_control_mrr]
+
+serve_headers = any value and key-pair present if the peer can serve header chain downloads.
+serve_chain_since = present if the peer can serve Body/Receipts ODR requests starting from the given block number.
+serve_state_since = present if the peer can serve Proof/Code ODR requests starting from the given block number.
+tx_relay =  present if the peer can relay transactions to the network.
+flow_control_bl = max credits (positive integer describing the burst-depth of the token bucket),
+flow_control_mrc = cost_table (see below)
+flow_control_mrr = rate of recharge (positive integer of credits recharged per second)
+```
+
+#### Cost table
+
+The cost table includes a mapping of individual [PIP message request IDs](#PIP-messages) to costs, which are applied in the token-bucket rate limiter. The [Headers](#headers) and [Execution](#execution) request messages are special cases where the cost is multiplied by the maximum number of requested header or gas requested, respectively. The table also includes a base cost, which is applied to the [request batch message](#request-message).
+
+```text
+cost_table = [base_cost, [id,cost],...]
+
+base_cost = positive integer cost applied to a request batch.
+id = identifier of an individual PIP message type
+cost = positive integer to apply to cost calculations for this message type
+```
+
+### Announcement
+
+`message-id: 0x01`
+
+```text
+message-data = message-id || [head_blockhash, head_blocknum, total_difficulty, reorg_depth, [key0,value0],[key1,value1],...]
+
+reorg_depth: positive integer containing the reorganization depth to the common ancestor of the new head and the last announced head.
+```
+
+The message element meanings are the same as in the [Status](#Status) with the exception of reorg_depth.
+
+## PIP Requests and Responses
+
+PIP request and response messages are batched and cannot be sent individually. Unlike LES, PIP batches may contain messages of different types.
+
+PIP tries to further optimise client-server round trips by allowing the individual requests in the batch to include references to what their responses would contain if processed sequentially. For clarification, an exampple PIP batch request could contain two request messages in order, where the second message specifies that an input is a specific 'output' of the first message, where 'output' means the server response to that request.
+
+### Request message
+
+`message-id: 0x02`
+
+```text
+message-data = message-id || [request-id, [req1, ...]]
+
+request-id = a unique scalar request identifier for request-reply correlation
+[req1, ...] = an rlp list of requests, each request being the rlp encoding of a request as specified below
+```
+
+### Response message
+
+`message-id: 0x03`
+
+```text
+message-data = message-id || [request-id, cr, [resp1, ...]]

```

Comment:
Instead of defining `message-data` for every message, you can just put the data list. We are using the same convention in all the other specs.